code/modules/mob/living/stats.dm

```diff
@@ -36,24 +36,16 @@
 	return TRUE
 
 /datum/species
-	var/list/specstats = list("strength" = 0, "perception" = 0, "intelligence" = 0, "constitution" = 0, "endurance" = 0, "speed" = 0, "fortune" = 0)
-	var/list/specstats_f = list("strength" = 0, "perception" = 0, "intelligence" = 0, "constitution" = 0, "endurance" = 0, "speed" = 0, "fortune" = 0)
-
-/mob/living/proc/roll_stats()
-	STASTR = 10
-	STAPER = 10
-	STAINT = 10
-	STACON = 10
-	STAEND = 10
-	STASPD = 10
-	STALUC = 10
-	for(var/S in MOBSTATS)
+	///Statkey = bonus stat, - for malice.
+	var/list/specstats = list(STATKEY_STR = 0, STATKEY_PER = 0, STATKEY_END = 0,STATKEY_CON = 0, STATKEY_INT = 0, STATKEY_SPD = 0, STATKEY_LCK = 0)
+	///Statkey = bonus stat, - for malice.
+	var/list/specstats_f = list(STATKEY_STR = 0, STATKEY_PER = 0, STATKEY_END = 0,STATKEY_CON = 0, STATKEY_INT = 0, STATKEY_SPD = 0, STATKEY_LCK = 0)
+
+///Rolls random stats base 10, +-2, for SPECIAL, and applies species stats and age stats.
+/mob/living/proc/roll_mob_stats()
+	for(var/stat in MOBSTATS)
 		if(prob(33))
-			switch(pick(1,2))
-				if(1)
-					change_stat(S, 1)
-				if(2)
-					change_stat(S, -1)
+			change_stat(stat,rand(-2,2))
```

Comment:
this functions differently before, first its -2, 2 instead of 1 or -1 it also can just roll 0 which negates the prob.

Reply:
Fixed Next Commit